### PR TITLE
log viewer performance #2163

### LINKED ISSUE #2204: plot is wrong when hand picking data in log viewer
If you select the rows one by one the points are ordered by the order you select them, not by time as expected.

One line fix on the way... :wink:

### PR DESCRIPTION
**Pull request by griebd**
Fixes #2204 and #2199 on next! (already fixed on master)

Try to solve #2163 on next. (easily to port back to master if accepted)

Updated QCustomPlot to latest version 1.3.0 (December 27, 2014). And put it in a sub-directory.

Follow suggestions from QCustomPlot to improve performance:
- pen width <= 1.0 (was 1.5)
- setNoAntialiasingOnDrag to true (as name suggests, disable anti-aliasing on drag only)

Function moveLegend() is commented out as it's a bit more work to migrate to new QCustomPlot and I couldn't see it working in the old version, so there is no lose of functionality... (correct me if I'm wrong)

When opening a new file:
- try to improve performance while loading the data on ui->logTable. There is some improvement, but unnoticeable. So far the only solution I did find is to change from QTableWidget to QTableView and that means a bigger change. And that is why I decided to put all this in next and them start working on this last change.
- improved performance on columns resize by changing the way its done.
- changed the parsing of log files completely. NEEDS TEST as I have only a couple of test files!!!! (but reduces the loading time by 50%)

Changed the way fly sessions are handled and now you get no loading time when you change the fly session (but if you have some fields selected the plotting time still exists).

Many changes, I know. So, review and tell me what needs to be fixed or changed or whatever.... :wink:
At least for me, it's a lot faster to use "log viewer" now... I hope everyone else feels the same!!!!

### CODE REVIEW

File: companion/src/logsdialog.cpp [hunk 1]
```diff
@@ -399,24 +407,25 @@ void logsDialog::removeAllGraphs()
...
-    bool ok;
-    int dataInt = contextAction->data().toInt(&ok);
-    if (ok)
-    {
-      ui->customPlot->legend->setPositionStyle((QCPLegend::PositionStyle)dataInt);
-      ui->customPlot->replot();
-    }
-  }
-}
+// void logsDialog::moveLegend()
```
Comment (bsongis) [inline companion/src/logsdialog.cpp]: Please remove the code if needed, unless it's a TODO and you keep the code for later?

File: companion/src/qcustomplot/GPL.txt [hunk 2]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (bsongis) [inline companion/src/qcustomplot/GPL.txt]: Was it a thirdparty? If yes it should be in its own directory (as I did for FatFS / Lua / CoOS / etc.), no?

Comment (griebd) [inline companion/src/qcustomplot/GPL.txt]: yes, it is a third party lib, and it's now in its own directory. (the last version wasn't)

File: companion/src/logsdialog.cpp (hunk 1 above)
Comment (griebd) [inline companion/src/logsdialog.cpp]: well, thing is I don't know if it's a TODO... will remove then, if someone finds the need I implement it later!

**Comment by bsongis** [comment]
Thanks a lot for this one!

**Pull request merged**